### PR TITLE
Azure Redis .Net SDK update for new api v2023-08-01

### PR DESCRIPTION
**Pull request by koderjoker**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

Swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/redis/resource-manager/Microsoft.Cache/stable/2023-08-01/redis.json

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by pallavit** [comment]
@koderjoker this PR is getting tagged as a community contribution. Please review the Azure SDK onboarding documentation and ensure that your GitHub organization membership and permissions are correct.

[Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) (Microsoft internal)
[Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) (Microsoft internal)

/cc: @ArthurMa1978. Also please ensure you have the correct aliases are added to the code owners file - https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS

**Comment by koderjoker** [comment]
@pallavit no new names to add to Code Owners from our side

**Comment by koderjoker** [comment]
Test recording in progress, will push session records once complete

**Comment by koderjoker** [comment]
Recordings updated as part of https://github.com/Azure/azure-sdk-assets/releases/tag/net%2Fredis%2FAzure.ResourceManager.Redis_50c384bcc2

**Review by jsquire — DISMISSED** [review]
@koderjoker:  You are still coming up as having none of the public organizations required nor write permissions in the repository.   Please review the Azure SDK onboarding guide. 
  This will need to be corrected before this PR can move forward.

**Comment by koderjoker** [comment]
@jsquire I'm already a part of Azure GitHub and Azure SDK partners. Applied for write permissions for Azure cache for Redis SDKs.

**Comment by koderjoker** [comment]
Hi @archerzz @jsquire @tg-msft @KrzysztofCwalina @ArcturusZhang @pallavit @ArthurMa1978 @live1206! Who would be the assigned reviewer for this PR?

Despite the PR having the latest code generated and recordings for 2023-08-01 https://github.com/Azure/azure-sdk-assets/releases/tag/net%2Fredis%2FAzure.ResourceManager.Redis_50c384bcc2, the pipeline is showing errors regarding both

**Review by archerzz — CHANGES_REQUESTED** [review]

**Comment by koderjoker** [comment]
> > Hi @archerzz @jsquire @tg-msft @KrzysztofCwalina @ArcturusZhang @pallavit @ArthurMa1978 @live1206! Who would be the assigned reviewer for this PR?
> > Despite the PR having the latest code generated and recordings for 2023-08-01 [Azure/azure-sdk-assets@`net%2Fredis%2FAzure.ResourceManager.Redis_50c384bcc2` (release)](https://github.com/Azure/azure-sdk-assets/releases/tag/net%2Fredis%2FAzure.ResourceManager.Redis_50c384bcc2), the pipeline is showing errors regarding both
> 
> @koderjoker I think the error message is quite clear:
> 
> ```
> Generated code is not up to date.
>     You may need to rebase on the latest main, 
>     run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
>     run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
>     run 'dotnet build /t:GenerateCode' to update the generated code and samples.
>     
> To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory redis'.
> ```
@archerzz running `eng\scripts\CodeChecks.ps1 -ServiceDirectory redis` as mentioned in the error, on our local gives us 0 errors. 

No new code is generated when `eng\scripts\Export-API.ps1` or `dotnet build /t:GenerateCode` are run, and no snippets on running `eng\scripts\Update-Snippets.ps11

**Comment by koderjoker** [comment]
@jsquire corrected

**Comment by jsquire** [comment]
> [koderjoker](/koderjoker)

Thanks.  Confirmed that we're looking good.   Removing my block.   Please work with Arthur and team on the code review.

File: sdk/redis/Azure.ResourceManager.Redis/src/Generated/RedisCacheAccessPolicyAssignmentData.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            ProvisioningState = provisioningState;
+            ObjectId = objectId;
+            ObjectIdAlias = objectIdAlias;
+            AccessPolicyName = accessPolicyName;
+        }
+
+        /// <summary> Provisioning state of an access policy assignment set. </summary>
+        public AccessPolicyAssignmentProvisioningState? ProvisioningState { get; }
+        /// <summary> Object Id to assign access policy to. </summary>
+        public string ObjectId { get; set; }
```
[the remark this answers is not in the document]
Comment (koderjoker) [reply sdk/redis/Azure.ResourceManager.Redis/src/Generated/RedisCacheAccessPolicyAssignmentData.cs]: Updated as GUID

File: sdk/redis/Azure.ResourceManager.Redis/src/Generated/Models/RedisCommonConfiguration.cs [hunk 2]
```diff
@@ -95,6 +97,8 @@ internal RedisCommonConfiguration(bool? isRdbBackupEnabled, string rdbBackupFreq
         public string AuthNotRequired { get; set; }
         /// <summary> SubscriptionId of the storage account for persistence (aof/rdb) using ManagedIdentity. </summary>
         public string StorageSubscriptionId { get; set; }
+        /// <summary> Specifies whether AAD based authentication has been enabled or disabled for the cache. </summary>
+        public string AadEnabled { get; set; }
```
[the remark this answers is not in the document]
Comment (koderjoker) [reply sdk/redis/Azure.ResourceManager.Redis/src/Generated/Models/RedisCommonConfiguration.cs]: Renamed as IsAadEnabled

**Comment by koderjoker** [comment]
Updated session records: https://github.com/Azure/azure-sdk-assets/tree/net/redis/Azure.ResourceManager.Redis_73d38d0d17/net/sdk/redis/Azure.ResourceManager.Redis/tests/SessionRecords

**Review by archerzz — APPROVED** [review]
@koderjoker LGTM.

By the way, if you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

**Review by archerzz — APPROVED** [review]

**Pull request merged**